lib/commons/dom/create-grid.js

```diff
@@ -323,36 +316,75 @@ function findScrollRegionParent(vNode, parentVNode) {
  * @param {VirtualNode}
  */
 function addNodeToGrid(grid, vNode) {
-  const gridSize = constants.gridSize;
   vNode.clientRects.forEach(rect => {
-    if (rect.right <= 0 || rect.bottom <= 0) {
-      return;
-    }
     // save a reference to where this element is in the grid so we
     // can find it even if it's in a subgrid
     vNode._grid ??= grid;
-    const x = rect.left;
-    const y = rect.top;
+    const gridRect = grid.getPositionOfRect(rect);
+    grid.loopGridPosition(gridRect, gridCell => {
+      if (!gridCell.includes(vNode)) {
+        gridCell.push(vNode);
+      }
+    });
+  });
+}
 
-    // "| 0" is a faster way to do Math.floor
-    // @see https://jsperf.com/math-floor-vs-math-round-vs-parseint/152
-    const startRow = (y / gridSize) | 0;
-    const startCol = (x / gridSize) | 0;
-    const endRow = ((y + rect.height) / gridSize) | 0;
-    const endCol = ((x + rect.width) / gridSize) | 0;
+class Grid {
+  constructor(container = null) {
+    this.container = container;
+    this.cells = [];
+  }
 
-    grid.numCols = Math.max(grid.numCols ?? 0, endCol);
+  toGridIndex(num) {
+    return Math.floor(num / constants.gridSize);
+  }
 
-    for (let row = startRow; row <= endRow; row++) {
-      grid.cells[row] = grid.cells[row] || [];
+  getCellFromPoint({ x, y }) {
+    const rowIndex = this.toGridIndex(y);
+    const colIndex = this.toGridIndex(x);
+    if (!isPointInRect({ y: rowIndex, x: colIndex }, this.boundaries)) {
```

Comment:
I think either the `pointInRect` function or this one should error or something if `this.boundries` is not set. This can happen when you can `getCellFromPoint` before calling `loopGridPosition` (which setups up the grid and boundries). Should inform when you call things out of order so we don't run into the problem in the future.

Or better yet, return early and return `null` if `this.boundaries` does not exist.